openinsteam-bg.js

```diff
@@ -0,0 +1,32 @@
+async function getCurrentTabUrl() {
+  return new Promise((resolve, reject) => {
+    browser.tabs
+      .query({ currentWindow: true, active: true })
+      .then((tabs) => {
+        const currentUrl = tabs[0].url;
+        return resolve(currentUrl);
+      })
+      .catch(reject);
+  });
+}
+
+const urlMatches = browser.runtime.getManifest().content_scripts[0].matches;
+
+browser.menus.create({
+  id: "open-in-steam",
+  title: "Open in Steam",
+  contexts: ["all"],
+  documentUrlPatterns: urlMatches,
+});
+
+browser.menus.onClicked.addListener(function (info) {
+  if (info.menuItemId == "open-in-steam") {
+    getCurrentTabUrl()
```

Comment:
Is there a reason we can't use `window.location.href` and then don't need the tabs permission?

Reply:
I couldn't get it to work. It seems that `window.location.href` will return the 'generated background page' for the extension.
`moz-extension://81889618-0325-4704-a81f-4cb7ef3c2a3a/_generated_background_page.html`

Reply:
Probably because it's running as a background script, I think you can just add it as a normal script then?

Reply:
I tried a bit but couldn't seem to get context menus to work from content scripts because the context does not exist in a content script unless you create a custom UI and open the context menu from within said AI.

Reply:
Okay, thanks for giving it a go. 👍